### PR TITLE
I423 - Update openjp2 library to v2.3.0 from v2.1.2

### PR DESCRIPTION
**Pull request by tst-eclamar**
Fixes: #423 
Update openjp2 library from v2.1.2 to v.2.3.0
* Create new library to replace old
* Modify SConscript to point to new library version
* Modify gdal configure script to find new location for header file

### CODE REVIEW

**Comment by tst-ccamp** [comment]
The changes made to the GDAL library should be done through the patch process like the various other patches we make to GDAL. The reason for this is so that they don't have to keep being made manually every time we do minor updates to the library. Please look into making the required changes using a patch. This isn't an urgent change, so we might as well do it right the first time around instead of as a follow-up.

**Comment by tst-nfarah** [comment]
Doing a test to compare openJpeg 2.1.2 vs openJpeg 2.3.0 trough 'gdal_translate' of two jp2000 files we have in the tutorial (bluemarble_4km.jp2 and usgsLanSat.jp2) to Geotiff, i got (on Ubuntu 16.04) the following:
bluemarble_4km.jp2: 
------------------------
OpenJpeg 2.1.2: 1381 seconds. 23 minutes
OpenJpeg 2.3.0: 20 seconds

usgsLanSat.jp2:
-----------------
OpenJpeg 2.1.2: 1370 seconds. 22.8 minutes
OpenJpeg 2.3.0:  23 seconds

These numbers are pretty good. 

This is said, it appears that we may need to patch GDAL a little bit more to be in sync with enabling multi-threading when decoding jp2 files. See last two revisions made to 'openjpegdataset.cpp':
[https://trac.osgeo.org/gdal/log/trunk/gdal/frmts/openjpeg/openjpegdataset.cpp?rev=40607](url)

as well as revision made part of this ticket: [https://trac.osgeo.org/gdal/ticket/7074](url)

Eventually we would upgrade our GDAL and get these changes, but for the time being we, probably, need to address them through a patch.

**Comment by tst-nfarah** [comment]
@tst-eclamar : I'm fine with creating a follow up ticket to patch GDAL more as mentioned in my previous comment. It appears that we get some performance results, but i think we can leverage multi-threaded decoding more by enabling it. The way to enable it is to patch 'openjpegdataset.cpp' though an API call to OpenJpeg library.

File: earth_enterprise/src/third_party/gdal/SConscript [hunk 1]
```diff
@@ -26,26 +26,11 @@
 # lines below.
 
 Import('third_party_env')
-import os
```
Comment (tst-nfarah) [inline earth_enterprise/src/third_party/gdal/SConscript]: These mrSID changes are now in master. Eventually you want to get them by updating your fork  or just ensuring that we get them when this PR is merged.

**Comment by tst-ccamp** [comment]
Changes look good and I see 2 approvals, so I'm merging this

**Pull request merged**